### ISSUE TITLE
no matching ID found for prefix

### ISSUE THREAD

**Issue by prabhinmp** [vmware-tanzu/velero#4205]
Hi Team,
Im using velero , restic and minio for backup in the onpremise environment.

I deleted backup of a particular service in default namespace after that scheduled backups are not happening. getting below error,

`me="2021-09-30T07:00:23Z" level=error msg="Error backing up item" backup=velerodaily-backup2-20210930070021 error="pod volume backup failed: error running restic backup, stderr=Fatal: invalid id \"5c38d1c5\": no matching ID found for prefix \"5c38d1c5\"\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/backupper.go:179" error.function="github.com/vmware-tanzu/velero/pkg/restic.(*backupper).BackupPodVolumes" logSource="pkg/backup/backup.go:431" name=-es-master-0`

**Comment by Leoscva** [comment]
Hi,

try upgrading to latest 1.6 version. In my case it solved the problem. Probably 1.7 will work also.

**Comment by prabhinmp** [comment]
Im using velero version 1.6.5.  Still I'm facing the issue

**Comment by tim-sendible** [comment]
I also experienced it today. I believe this happens when restic gets confused and one of the previous backups it relies upon (restic does incremental backups) disappears or gets corrupted somehow. The (horrific) workaround I have found is to either delete all the backups in s3 or to change the s3 prefix forcing Velero to basically start again. This only works if there isn't a backup or deletion job mid-flight... took me a few attempts to get it right.

This is absolutely not a sensible or production-ready fix... but hopefully it helps to pinpoint the root cause.

**Comment by blackpiglet** [comment]
Thanks @tim-sendible.
I agree with your analysis. Restic use an incremental backup strategy. If the previous manual backup is the base of the scheduled backup, and the previous manual backup is deleted, then the scheduled backup should face a failure. 

@prabhinmp 
I'm new to Velero project. I cannot assure that previous restic backup deletion is not allowed anyway.
I think to, let the restic backup work again, you can delete the original ResticRepository, then create a new one with different S3 location. By this way, restic should do a total backup from beginning.

**Comment by ugur99** [comment]
Hi @blackpiglet we're experiencing same [problem](https://github.com/vmware-tanzu/velero/issues/2507#issuecomment-1535995921) and I deleted related `backuprepositories` object and its s3 bucket, but its still getting the same error. Do you have any other suggestions as a workaround.

**Comment by blackpiglet** [comment]
@ugur99 
I tested with Velero v1.11 with a clean bucket, and set the prefix to `demo`. It worked.
Could you check whether the BackupStorageLocation and the bucket setting match?

`no matching ID found for prefix "f07ec467"`
I think the error is caused by Restic cannot find snapshots according to the prefix setting. Is there a directory named `f07ec467` in your bucket, after the Velero installation?

**Comment by ugur99** [comment]
Thank you for quick response @blackpiglet, I deleted all `podvolumebackups` (regarding the successfull and failed ones), also deleted `backuprepositories` object of the related ns and lastly deleted the empty s3 bucket related to that ns and it worked! When updating velero to the last major version we've changed `bucket_name` and after then some of the Restic backups failed for this reason; but I think these steps solved the issue. Thank you for your help!
